nabo/index_heap.h

```diff
@@ -73,7 +73,12 @@ namespace Nabo
 		typedef typename Eigen::Matrix<Index, Eigen::Dynamic, 1> IndexVector;
 		//! vector of values
 		typedef typename Eigen::Matrix<Value, Eigen::Dynamic, 1> ValueVector;
-		
+
+		//! invalid index
+		static constexpr auto InvalidIndex = std::numeric_limits<IT>::max();
```

Comment:
Once you've converted this to a templated constexpr inline function, you could return as value : 

```c++
std::is_unsigned<IT>::value ? std::numeric_limits<IT>::max() : IT(-1); 
```